src/main/java/org/commonjava/indy/service/archive/services/LogService.java

```diff
@@ -0,0 +1,276 @@
+                        break;
+                    case "uploads":
+                        log.setUploads(readTraceLogItemsArray(reader));
+                        break;
+                    case "downloads":
+                        log.setDownloads(readTraceLogItemsArray(reader));
+                        break;
+                    default:
+                        reader.skipValue(); //avoid some unhandled events
+                        break;
+                }
+            }
+            reader.endObject();
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return log;
+    }
+
+    private Map<String,String> readTraceLogKey(JsonReader reader) throws IOException {
+        Map<String,String> key = new HashMap<String,String>();
+        while (reader.hasNext()) {
+            reader.beginObject();
+            while (reader.hasNext()) {
+                key.put(reader.nextName(),reader.nextString());
+            }
+            reader.endObject();
+        }
+        return key;
+    }
+
+    private List<TraceLogItem> readTraceLogItemsArray(JsonReader reader) throws IOException {
+        List<TraceLogItem> items = new ArrayList<TraceLogItem>();
+
+        reader.beginArray();
+        while (reader.hasNext()) {
+            reader.beginObject();
+            while (reader.hasNext()) {
+                items.add(readTraceLogItems(reader));
+            }
+            reader.endObject();
+        }
+        reader.endArray();
+
+        return items;
+    }
+
+    private TraceLogItem readTraceLogItems(JsonReader reader) throws IOException {
+        TraceLogItem item = new TraceLogItem();
+
+        while (reader.hasNext()){
+            String jsonKey = reader.nextName();
+            switch (jsonKey) {
+                case "storeKey":
+                    item.setStoreKey(reader.nextString());
+                    break;
+                case "accessChannel":
+                    item.setAccessChannel(reader.nextString());
+                    break;
+                case "path":
+                    item.setPath(reader.nextString());
+                    break;
+                case "localUrl":
+                    item.setLocalUrl(reader.nextString());
+                    break;
+                case "md5":
+                    item.setMd5(reader.nextString());
+                    break;
+                case "sha256":
+                    item.setSha256(reader.nextString());
+                    break;
+                case "sha1":
+                    item.setSha1(reader.nextString());
+                    break;
+                case "size":
+                    item.setSize(reader.nextLong());
+                    break;
+                default:
+                    reader.skipValue();
+                    break;
+            }
+        }
+
+        return item;
+    }
+
+
+
+    public static class TraceLog{
```

Comment:
Recommend to extract this as a separate class but not embedded in other class. And seems this is indy's folo tracking content, so recommend to use "TrackedContent" as class name to apply same with indy. See: https://github.com/Commonjava/indy/blob/master/addons/folo/model-java/src/main/java/org/commonjava/indy/folo/model/TrackedContent.java

Reply:
do I import of just copy the src

Reply:
This is done in https://github.com/Commonjava/indy-archive-service/pull/5